### PR TITLE
fix: battle navigation

### PR DESCRIPTION
**Pull request by IskandarGhobril**
**Summary:**  
This PR introduces minor adjustments to the navigation flow between battle screens, preparing the codebase for an upcoming bug fix and future enhancements.

**Changed Files:**

1. **`WaitingForCompletionScreen.kt`**: Updated composable arguments to use `friendUid` instead of `userUid` to better support future features.
2. **`MainActivity.kt`**: Added necessary logic to pass updated arguments for navigation.
3. **`BattleViewModel.kt`**: Added a line to retrieve the `friendUid` to pass it as an argument before navigating to the waiting screen.
4. **`NavigationActions.kt`**: Adjusted navigation methods to align with updated navigation paths.
5. **`WaitingForCompletionScreenTest.kt`**: Updated test cases to match the new composable signature and ensure test integrity.

### CODE REVIEW

**Review by MathisHage — APPROVED** [review]
The changes improve the clarity of the meaning of some variables pretty well. Approved.

File: app/src/androidTest/java/com/github/se/orator/ui/battle/WaitingForCompletionScreenTest.kt [hunk 1]
```diff
@@ -148,32 +149,32 @@ class WaitingForCompletionScreenTest {
...
     verify(mockNavigationActions, never()).navigateToEvaluationScreen(anyString())
   }
 
   /**
-   * Test navigation when the other user completes the battle. We simulate a battle update where the
+   * Test navigation when the friend completes the battle. We simulate a battle update where the
    * opponentCompleted is now true.
    */
   @Test
   fun testNavigationWhenOtherUserCompleted() {
```
Comment (Rolph1) [inline app/src/androidTest/java/com/github/se/orator/ui/battle/WaitingForCompletionScreenTest.kt:161]: Maybe split this into different sub tests so that it's easier to see when one button stopped working

**Review by Rolph1 — APPROVED** [review]
## Overall:
Great PR that enhances not only the navigation, but also the tests! You also worked on fixing the documentation of the tests, something that should be done more often in this project.

## Suggested improvements:
- Whenever you make new tests/enhance them, it would be nice to add those to the End to end test. I am currently working on adding old screens and will be adding these tests myself so I won't request changes, but please keep this in mind next time you make new tests.

## In conclusion:
Great work Iskandar! I am also approving this PR. Please don't forget to add your tests to end to end next PR!

File: app/src/androidTest/java/com/github/se/orator/ui/battle/WaitingForCompletionScreenTest.kt (hunk 1 above)
Comment (IskandarGhobril) [reply app/src/androidTest/java/com/github/se/orator/ui/battle/WaitingForCompletionScreenTest.kt:161]: I believe there might be a misunderstanding here. This test is focused on checking the navigation and doesn't cover component display verification. Could you clarify if that's what you meant?

File: app/src/androidTest/java/com/github/se/orator/ui/battle/WaitingForCompletionScreenTest.kt [hunk 2]
```diff
@@ -148,32 +149,32 @@ class WaitingForCompletionScreenTest {
...
     composeTestRule.onNodeWithTag("loadingIndicator").assertIsDisplayed()
 
-    // No navigation should have happened yet since otherUser not completed
+    // No navigation should have happened yet since friend has not completed
     verify(mockNavigationActions, never()).navigateToEvaluationScreen(anyString())
   }
 
   /**
-   * Test navigation when the other user completes the battle. We simulate a battle update where the
+   * Test navigation when the friend completes the battle. We simulate a battle update where the
```
[the remark this answers is not in the document]
Comment (IskandarGhobril) [reply app/src/androidTest/java/com/github/se/orator/ui/battle/WaitingForCompletionScreenTest.kt]: Nice catch! I fixed it in my most recent push

File: app/src/androidTest/java/com/github/se/orator/ui/battle/WaitingForCompletionScreenTest.kt [hunk 3]
```diff
@@ -93,10 +94,10 @@ class WaitingForCompletionScreenTest {
     }
 
     // Mock getUserProfile to return a valid user profile
-    `when`(mockUserProfileRepository.getUserProfile(eq("testUser"), any(), any())).thenAnswer {
+    `when`(mockUserProfileRepository.getUserProfile(eq(currentUserId), any(), any())).thenAnswer {
```
[the remark this answers is not in the document]
Comment (IskandarGhobril) [reply app/src/androidTest/java/com/github/se/orator/ui/battle/WaitingForCompletionScreenTest.kt:97]: Thanks!

**Review by Rolph1 — APPROVED** [review]
Great work Iskandar! Everything seems good to me now. I am approving this PR!
